OSOE-1010: Update to Node.js v24

Piedone opened the pull request:
[OSOE-1010](https://lombiq.atlassian.net/browse/OSOE-1010)
Fixes #965

[OSOE-1010]: https://lombiq.atlassian.net/browse/OSOE-1010?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Piedone committed:
Build passes but due to pnpm linking not working, it doesn't actually run node builds

Piedone committed:
Pointing GHA to issue branch

Piedone committed:
Submodule src/Utilities/Lombiq.NodeJs.Extensions updated

Submodule src/Utilities/Lombiq.NodeJs.Extensions:
    > d1905aa - Using pnpm link that actually links
    > f4ab919 - While @eslint/eslintrc is not supposed to be needed directly, without it, we get "ERROR: Cannot find module '@eslint/eslintrc'"
    > d32b747 - Docs
    > f6e94d3 - Docs

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-1010

# Conflicts:
#	.github/workflows/build-and-test-windows.yml
#	.github/workflows/build-and-test.yml

Piedone committed:
Downgrading PNPM to 9.15.9

Piedone committed:
NuGet publish

Piedone committed:
Submodule src/Utilities/Lombiq.NodeJs.Extensions updated

Submodule src/Utilities/Lombiq.NodeJs.Extensions:
    > 343f854 - We need a direct @eslint/eslintrc reference in NuGet consumers

Piedone committed:
Submodule src/Modules/Lombiq.DataTables updated

Submodule src/Modules/Lombiq.DataTables:
    > 5e7becc - Fixing that the DataTables CDN version wasn't updated and removing CDN integrity since that can't easily be kept up-to-date with it

Piedone committed:
Missing package lock update

sarahelsaig reviewed (changes requested):
There are merge conflicts.

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-1010

Piedone committed:
Submodule src/Utilities/Lombiq.NodeJs.Extensions updated

Submodule src/Utilities/Lombiq.NodeJs.Extensions:
    > 4f82947 - Adding missing node_modules excludes

sarahelsaig committed:
submodules

sarahelsaig committed:
revert GHA branches

Piedone committed:
Pointing GHA to issue branch

Piedone committed:
Submodule src/Utilities/Lombiq.NodeJs.Extensions updated

Submodule src/Utilities/Lombiq.NodeJs.Extensions:
    > bb696ac - Docs
    > 857b202 - Fix PNPM setup
    > a60c882 - Path to package.json
    > 615dfb6 - Trying pnpm/action-setup
    > 6911c82 - Trying to set the PATH from a variable instead
    > 4b5be14 - Manual folder
    > 0bef8c3 - Revert "Debug"
    > 92d5105 - Debug
    > f508e3a - Trying manual fix
    > b05374f - Debug
    > 854921a - Maybe the Linux check is incorrect
    > 714f2d7 - Fixing pnpm setup
    > 1c5509c - Running pnpm setup
    > f01ed32 - Enabling corepack and setting up Node in the workflow

Piedone committed:
Running the GitHub-hosted builds temporarily too

Piedone committed:
Dummy change to kick off build

Piedone committed:
Revert "Dummy change to kick off build"

This reverts commit b46bb9182eeea97a27a3f661752e14c56caf7f18.

Piedone committed:
Submodule src/Utilities/Lombiq.NodeJs.Extensions updated

Submodule src/Utilities/Lombiq.NodeJs.Extensions:
    > 1a8e808 - Fully qualifying package_json_file

Piedone committed:
Submodule src/Utilities/Lombiq.NodeJs.Extensions updated

Submodule src/Utilities/Lombiq.NodeJs.Extensions:
    > 7f63018 - Code styling

Piedone committed:
Submodule src/Utilities/Lombiq.NodeJs.Extensions updated

Submodule src/Utilities/Lombiq.NodeJs.Extensions:
    > d1d71fe - Need resolved NE path when setting up PNPM

Piedone committed:
Submodule src/Utilities/Lombiq.NodeJs.Extensions updated

Submodule src/Utilities/Lombiq.NodeJs.Extensions:
    > 9024545 - Removing debug code
    > cca3792 - NE idiosynchraciv
    > 547eace - Feeding the PNPM version to pnpm/action-setup
    > d2eef15 - Debug
    > ecd52e3 - Revert "Maybe we need the subfolder"
    > 497b95e - Maybe we need the subfolder
    > 79242e6 - Code styling

Piedone committed:
Revert "Running the GitHub-hosted builds temporarily too"

This reverts commit 0b2e12d7dba9d625651bcf165047214a276c4a08.

[skip ci]

Piedone committed:
Reverting issue branch references for GHA

Piedone committed:
NuGet publish

Piedone committed:
Updating submodules to merged dev

[skip ci]
